src/search/_Search.jsx

```diff
@@ -0,0 +1,154 @@
+  disclosureManager.closeDisclosure();
+};
+
+const highlight = (key, result) => {
+  // Filter the matches to only ones that match the key
+  const valueArray = result.matches.filter(item => item.key === key).map((match) => {
+    let startIndex = 0;
+    // Split the value string to have spans around the matching indices
+    const splitString = match.indices.reduce((acc, index) => {
+      acc.push(match.value.slice(startIndex, index[0]));
+      acc.push(<span key={index[0]}>{match.value.slice(index[0], index[1] + 1)}</span>);
+      startIndex = index[1] + 1;
+      return acc;
+    }, []);
+    splitString.push(match.value.slice(startIndex));
+    return splitString;
+  });
+  // The item has a match that should be highlighted
+  if (valueArray.length === 1) {
+    return valueArray[0];
+  }
+  // else return the item
+  return result.item[key];
+};
+
+const searchItem = result => (
+  <div className={cx('item')}>
+    <div className={cx('title')}>
+      {highlight('title', result)}
+    </div>
+    <div className={cx('path')}>
+      {highlight('path', result)}
+    </div>
+  </div>
+);
+
+const cacheSearchItems = (fetchSearchItems, state, setState) => {
+  if (!state.searchItems) {
+    fetchSearchItems().then((searchItems) => {
+      const { results, searchString } = state;
+      setState({ results, searchString, searchItems });
+      handleSearch(searchString, state, setState);
+    });
+  }
+};
+
+const Search = ({ disclosureManager, fetchSearchItems, onItemSelected }) => {
+  const [state, setState] = useState({ results: [] });
+  cacheSearchItems(fetchSearchItems, state, setState);
+  const { searchItems, searchString, results } = state;
+  return (
+    <ContentContainer
+      header={(
+        <>
+          <ActionHeader
+            title="Site Search"
+            onBack={disclosureManager.goBack}
+            onClose={disclosureManager.closeDisclosure}
+          />
+          <SearchField
+            className={cx('search-field')}
+            isBlock
+            placeholder="Search"
+            onSearch={string => handleSearch(string, state, setState)}
+            onInvalidSearch={() => clearResults(setState)}
+          />
+        </>
+      )}
+      fill
+    >
+      {searchItems && searchString && results.length <= 0 && <StatusView variant="no-matching-results" />}
+      {results.length > 0 && (
+        <InfiniteList
+          dividerStyle="standard"
+        >
+          {
+            state.results.map(result => (
+              <Item
+                key={result.item.path}
+                isSelectable
+                metaData={result}
+                onSelect={(event, metaData) => handleSelect(metaData, onItemSelected, disclosureManager)}
+              >
+                {searchItem(result)}
+              </Item>
+            ))
+          }
+        </InfiniteList>
+      )}
+    </ContentContainer>
+  );
+};
+
+Search.propTypes = propTypes;
+
+export default withDisclosureManager(Search);
```

Comment:
You could use the DisclosureManagerContext directly here with useContext and lose the HOC